test/creds_test.go

```diff
@@ -430,19 +428,25 @@ func (s) TestCredsHandshakeAuthority(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	defer lis.Close()
+
 	cred := &authorityCheckCreds{}
 	s := grpc.NewServer()
 	go s.Serve(lis)
 	defer s.Stop()
 
 	r := manual.NewBuilderWithScheme("whatever")
 
-	cc, err := grpc.Dial(r.Scheme()+":///"+testAuthority, grpc.WithTransportCredentials(cred), grpc.WithResolvers(r))
+	r.InitialState(resolver.State{
+		Addresses: []resolver.Address{{Addr: lis.Addr().String()}},
+	})
```

Comment:
I believe you can keep the `r.UpdateState` call below and remove this one to maintain the old test behaviour. Since a `cc.Connect()` is called, the resolver should be built before calling `r.UpdateState`. Same comment for the next test.

Reply:
used r.UpdateState